src/utils/index.js

```diff
@@ -386,19 +426,22 @@ export const getFinalCropItemIdFromSeedItemId = (
   }
 }
 
-/**
- * @param {string} cropItemId
- * @returns {string}
- */
 export const getSeedItemIdFromFinalStageCropItemId = memoize(
-  cropItemId => {
-    return Object.values(itemsMap).find(({ growsInto }) => {
+  (/** @type {string} */ cropItemId) => {
+    const seedItemId = Object.values(itemsMap).find(({ growsInto }) => {
       if (Array.isArray(growsInto)) {
         return growsInto.includes(cropItemId)
       } else {
         return growsInto === cropItemId
       }
-    }).id
+    })?.id
+
+    if (!seedItemId)
+      throw new Error(
+        `Crop item ID ${cropItemId} does not have a corresponding seed`
+      )
```

Comment:
This error should never be thrown in practice, but it was necessary for this function to be type safe. I expect we'll need to use this pattern more as we adopt Typed JavaScript.